distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Widget_Helpers.enso

```diff
@@ -16,40 +18,48 @@ import project.Data.Table_Conversions
 make_aggregate_column_selector : Table -> Display -> Boolean -> Widget
 make_aggregate_column_selector table display=Display.Always include_group_by=True =
     col_names_selector = make_column_name_selector table display=Display.Always
-    column_widget = Pair.new "column" col_names_selector
+    column_widget = ["column", col_names_selector]
 
     col_list_selector = make_column_name_vector_selector table display=Display.Always
 
-    group_by = if include_group_by then [Option "Group By" "(Aggregate_Column.Group_By)" [column_widget]] else []
-    count = Option "Count" "Aggregate_Column.Count"
-    count_distinct = Option "Count Distinct" "(Aggregate_Column.Count_Distinct)" [Pair.new "columns" (col_list_selector)]
-    first = Option "First" "(Aggregate_Column.First)" [column_widget, Pair.new "order_by" (col_list_selector)]
-    last = Option "Last" "(Aggregate_Column.Last)" [column_widget, Pair.new "order_by" (col_list_selector)]
+    fqn = Meta.get_qualified_type_name Aggregate_Column
+    group_by = if include_group_by then [Option "Group By" fqn+".Group_By" [column_widget]] else []
+    count = Option "Count" fqn+".Count"
+    count_distinct = Option "Count Distinct" fqn+".Count_Distinct" [["columns", col_list_selector]]
+    first = Option "First" fqn+".First" [column_widget, ["order_by" , col_list_selector]]
+    last = Option "Last" fqn+".Last" [column_widget, ["order_by" , col_list_selector]]
 
-    count_not_nothing = Option "Count Not Nothing" "(Aggregate_Column.Count_Not_Nothing)" [column_widget]
-    count_nothing = Option "Count Nothing" "(Aggregate_Column.Count_Nothing)" [column_widget]
+    count_not_nothing = Option "Count Not Nothing" fqn+".Count_Not_Nothing" [column_widget]
+    count_nothing = Option "Count Nothing" fqn+".Count_Nothing" [column_widget]
 
     ## Should be a list of Text columns only
```

Comment:
Should we start implementing this by-type column filtering? We have all the capabilities needed to do that.

If not in this PR, I'd be happy to create a small PR for this - it's just a few lines of adjustment to get a really nice UX improvement.

Reply:
Oh, why?

Reply:
Hmm, I see. Well indeed maybe it would be less confusing if we chose the column _first_ and then had a list of aggregations available for it - but that does not fit our `Aggregate_Column` format.

IMHO suggesting invalid columns for an aggregation is not that helpful, I don't imagine it is a problem, but if that's the recommendation.

Reply:
I was thinking the same and wondered around adding a note if an unsupported type - so you would see the same set but something in brackets warning invalid for this).

Reply:
Yes, I think that would be good. Or even, if we had the ability to add styling, the invalid types could be greyed out.

Also, I would move all unsupported down to the bottom, so that the first fields that show up are the supported ones. No sense in having to scroll thru unsupported fields to find a good one.

Reply:
Shall we create a follow up task for this?

I think the sorting will be especially useful, and marking the not-matching ones should also be helpful. Since we don't have formatting yet, maybe we can mark them like `/Shortest/`?

Reply:
Will add a task.